services/datamanager/builtin/builtin.go

```diff
@@ -454,6 +457,8 @@ func (svc *builtIn) Reconfigure(
 	deps resource.Dependencies,
 	conf resource.Config,
 ) error {
+	g := utils.NewGuard(func() { goutils.UncheckedError(svc.Close(ctx)) })
```

Comment:
We call Close() on datamanager if Reconfigure returns an error (so that we terminate any goroutines started up until that point by Reconfigure.

Reply:
Dan wrote it :)